### DISCUSSION TITLE
Which dependencies work together?

### DISCUSSION THREAD

**Discussion by dsterry2000** [spring-io/start.spring.io#1385] [Q&A]
I generated a Maven project using the Spring Initializr website for Java 17, SpringBoot 3.2.1, added Spring Web and Thymeleaf. I am learning Spring using the book Spring - Start Here by Laurentiu Spilca. He has full code examples which work. But I am trying to code the examples myself. My code is exactly the same, but beginning with the examples using Thymeleaf, my code doesn't work. It seems like the Dispatcher Servlet can't locate the html template in the src/main/resources/templates folder. I know this is what the problem is because I can move the file to the static folder and it is displayed by the client. My pom.xml file looks just like his except for the SpringBoot version. What is going on? How do you know which dependencies to use together?

**Comment by snicoll** [comment]
Sorry but this is the wrong place to ask. If you need support with using Spring and Thymeleaf, please ask on StackOverflow. Sharing what you have built so far (by pushing it to a GitHub repository and sharing the link) will help those who are trying to help. Feel free to add a link here so that we can follow along.
